mempool/mock/mempool.go

```diff
@@ -8,17 +8,31 @@ import (
 )
 
 // Mempool is an empty implementation of a Mempool, useful for testing.
-type Mempool struct{}
+type Mempool struct {
+	txs        map[types.TxKey]types.Tx
+	evictedTxs map[types.TxKey]bool
+}
 
-var _ mempool.Mempool = Mempool{}
+var _ mempool.Mempool = &Mempool{}
 
 func (Mempool) Lock()     {}
 func (Mempool) Unlock()   {}
 func (Mempool) Size() int { return 0 }
-func (Mempool) CheckTx(_ types.Tx, _ func(*abci.Response), _ mempool.TxInfo) error {
+func (m *Mempool) CheckTx(tx types.Tx, _ func(*abci.Response), _ mempool.TxInfo) error {
+	if m.txs == nil {
+		m.txs = make(map[types.TxKey]types.Tx)
+	}
+	if m.evictedTxs == nil {
+		m.evictedTxs = make(map[types.TxKey]bool)
+	}
+	m.txs[tx.Key()] = tx
+	return nil
+}
+func (m *Mempool) RemoveTxByKey(txKey types.TxKey) error {
+	delete(m.txs, txKey)
+	m.evictedTxs[txKey] = true
 	return nil
 }
```

Comment:
[question] why does removing always add the tx to the evictedTxs? Can't a tx be removed from the mempool by accepting it in a block and not evicting it from the mempool?

Reply:
i only did this in the mock to test txstatus which now has been reverted